internal/provider/data_file_test.go

```diff
@@ -0,0 +1,107 @@
+package provider_test
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/neuspaces/terraform-provider-system/internal/acctest"
+	"github.com/neuspaces/terraform-provider-system/internal/acctest/tfbuild"
+	"testing"
+)
+
+func TestAccDataFile_read_content(t *testing.T) {
+	testConfig := newTestFileConfig()
+
+	acctest.Current().Targets.Foreach(t, func(t *testing.T, target acctest.Target) {
+		t.Parallel()
+
+		resource.Test(t, resource.TestCase{
+			ProviderFactories: acctest.ProviderFactories(),
+			Steps: []resource.TestStep{
+				{
+					Config: tfbuild.FileString(tfbuild.File(
+						acctest.ProviderConfigBlock(target.Configs.Default()),
+						testAccFileBlock("test", testRunFilePath(target, testConfig.fileName),
+							tfbuild.AttributeString("mode", "644"),
+							tfbuild.AttributeString("content", "hello world!"),
+						),
+						testAccDataFileBlock("test", testRunFilePath(target, testConfig.fileName)),
```

Comment:
The dependency of the `file` data source which subject to test on the `file` resource which create the file is not defined in the test configuration. Therefore, the `file` data source attempts to read a file which does not yet exist. 

Currently, the acceptance test is equivalent to the following Terraform config. From the snippet, the missing resource dependency is easier to spot.

```terraform
resource "system_file" "test" {
	path    = "/root/test99090b78/file-1"
	mode    = "644"
	content = "hello world!"
}

data "system_file" "test" {
	path = "/root/test99090b78/file-1"
}
```

The test will work using the following Terraform config.

```terraform
resource "system_file" "test" {
	path    = "/root/test99090b78/file-1"
	mode    = "644"
	content = "hello world!"
}

data "system_file" "test" {
	path = system_file.test.path
}
```

Other acceptance tests, for example `TestAccUser_create`, also need resource dependencies in the test configuration.

With the pattern to define the test config in the acceptance tests, it is more difficult than with plain Terraform. You can use the `tfbuild.AttributeTraversal` to configure `path = system_file.test.path` in the data source.

```suggestion
						tfbuild.Data("system_file", "test",
							tfbuild.AttributeTraversal("path", tfbuild.TraversalResourceAttribute("system_file", "test", "path")),
						),
```

Reply:
Thank you, this was the remaining issue locally (I already set id to path).